backend/src/bin/consumer/main.rs

```diff
@@ -0,0 +1,356 @@
+        .consumer()
+        .with_topic(topic)
+        .with_subscription_type(SubType::Exclusive)
+        .build()
+        .await?;
+
+    match create_typesense_collections().await {
+        Ok(_) => {
+            println!("Typesense successfully initialized");
+        }
+        Err(e) => {
+            panic!("Failed to initialize typesense: {}", e);
+        }
+    };
+    let client = reqwest::Client::new();
+    while let Some(msg) = consumer.try_next().await? {
+        consumer.ack(&msg).await?;
+        println!("metadata: {:?},id: {:?}", msg.metadata(), msg.message_id());
+        let data = match msg.deserialize() {
+            Ok(data) => data,
+            Err(e) => {
+                println!("could not deserialize message: {:?}", e);
+                continue;
+            }
+        };
+        // if data.data.as_str() != "data" {
+        //     println!("Unexpected payload: {}", &data.data);
+        //     break;
+        // }
+        match data {
+            PulsarSearchData::CreateBurrow(burrow) => {
+                let data: TypesenseBurrowData = burrow.into();
+
+                match client
+                    .build_post("/collections/burrows/documents")
+                    .body(serde_json::to_string(&data).unwrap())
+                    .send()
+                    .await
+                {
+                    Ok(r) => println!("add new burrow.{:?}", r),
+                    Err(e) => println!("add new burrow failed {:?}", e),
+                }
+            }
+            PulsarSearchData::CreatePost(post) => {
+                let data: TypesensePostData = post.into();
+
+                match client
+                    .build_post("/collections/burrows/documents")
+                    .body(serde_json::to_string(&data).unwrap())
+                    .send()
+                    .await
+                {
+                    Ok(r) => println!("add new post.{:?}", r),
+                    Err(e) => println!("add new post failed {:?}", e),
+                }
+            }
+            PulsarSearchData::CreateReply(reply) => {
+                let data: TypesenseReplyData = reply.into();
+
+                match client
+                    .post("/collections/burrows/documents")
+                    .body(serde_json::to_string(&data).unwrap())
+                    .send()
+                    .await
+                {
+                    Ok(r) => println!("add new reply.{:?}", r),
+                    Err(e) => println!("add new reply failed {:?}", e),
+                }
+            }
+            PulsarSearchData::UpdateBurrow(burrow) => {
+                // TODO: read from typesense first, check the time if it is newer than the one in typesense
+                let search_result:SearchResult = serde_json::from_str(&client
+                .build_get(&format!("/collections/burrows/documents/search?q={}&query_by=burrow_id&filter_by=&sort_by=",burrow.burrow_id))
```

Comment:
here you have the burrow id, I think you don't need to perform a search. Simply get the document.